File: src/main/java/com/sppascm/project/service/StudentService.java
package com.sppascm.project.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.sppascm.project.model.Student;
import com.sppascm.project.repository.StudentRepository;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService {

    @Autowired
    private StudentRepository studentRepository;

    public Student register(Student student) {
        return studentRepository.save(student);
    }

    public boolean login(String kuId, String password) {
        System.out.println("Login Attempt: " + kuId + " / " + password);
        Optional<Student> student = studentRepository.findByKuId(kuId);
        if (student.isPresent()) {
            System.out.println("Student Found: " + student.get().getKuId());
            System.out.println("Password in DB: " + student.get().getPassword());
        } else {
            System.out.println("Student Not Found");
        }
        return student.isPresent() && student.get().getPassword().equals(password);
    }
    

    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    // 🔁 Update password logic
    public boolean updatePassword(String kuId, String newPassword) {
        Optional<Student> studentOpt = studentRepository.findByKuId(kuId);
        if (studentOpt.isPresent()) {
            Student student = studentOpt.get();
            student.setPassword(newPassword);
            studentRepository.save(student);
            return true;
        }
        return false;
    }

    // ❌ Delete student logic
    public boolean deleteStudentByKuId(String kuId) {
        Optional<Student> studentOpt = studentRepository.findByKuId(kuId);
        if (studentOpt.isPresent()) {
            studentRepository.delete(studentOpt.get());
            return true;
        }
        return false;
    }
}

File: project-management-frontend/src/services/studentService.js
// src/services/studentService.js
import axios from 'axios';
import BASE_URL from './api'; // Make sure BASE_URL = "http://localhost:8080"

export const registerStudent = async (student) => {
  // POST request to register a student
  return axios.post(`${BASE_URL}/students/register`, student);
};

export const loginStudent = async (student) => {
  // POST request to login student
  return axios.post(`${BASE_URL}/students/login`, student);
};

export const updatePassword = async (kuId, newPassword) => {
  // PUT request to update password
  return axios.put(`${BASE_URL}/students/update-password/${kuId}`, {
    newPassword: newPassword,
  });
};

export const deleteStudent = async (kuId) => {
  // DELETE request to remove student by kuId
  return axios.delete(`${BASE_URL}/students/delete/${kuId}`);
};
